lib/index.ts

```diff
@@ -1,36 +1,36 @@
-import * as path from 'path';
-import * as fs from 'fs';
+import {parseConfig} from './parser';
+
+interface DepDict {
+  [dep: string]: DepTree;
+}
 
 export interface DepTree {
   name: string;
   version: string;
-  dependencies: {
-    [dep: string]: DepTree;
-  };
+  dependencies?: DepDict;
   depType?: 'prod' | 'dev';
   hasDevDependencies?: boolean;
   missingLockFileEntry?: boolean;
 }
 
+export type PackageManagerType = 'dep'|'govendor';
+
 export async function buildDepTree(
   manifestFileContents: string,
   lockFileContents: string,
-  includeDev = false,
-  lockfileType?: string,
-  strict = true,
-  defaultManifestFileName = 'Gopkg.toml',
-): Promise<DepTree> {
-
-  throw new Error('Not implemented');
-}
-
-export async function buildDepTreeFromFiles(
-  root: string,
-  manifestFilePath: string,
-  lockFilePath: string,
-  includeDev = false,
-  strict = true,
+  lockfileType: PackageManagerType,
 ): Promise<DepTree> {
-
-  throw new Error('Not implemented');
+  const config = parseConfig(lockfileType, manifestFileContents, lockFileContents);
```

Comment:
can you call this something other than `config`, just cause that is usually reserved for an actual config. `goConfig`?